libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -0,0 +1,443 @@
+import tempfile
+import subprocess
+import os
+import shutil
+import json
+import time
+import logging
+import threading
+import urllib.request
+import urllib.error
+
+from libpermian.plugins import api
+from libpermian.result import Result
+from libpermian.workflows.isolated import IsolatedWorkflow
+from libpermian.events.structures.base import BaseStructure
+from libpermian.plugins.compose import ComposeStructure
+from libpermian.exceptions import StructureConversionError
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@api.events.register_structure('InstallationSource')
+class InstallationSourceStructure(BaseStructure):
+    """ Structure with URLs and paths needed for compose installation
+
+    Example:
+    {"base_repo": "BaseOS",
```

Comment:
As mentioned below, the `base_repo` item name should be renamed to `base_repo_id` as it stores just the ID and not URL of a repo.